### PR TITLE
google-protobuf - added built-in proto classes

### PR DESCRIPTION
**Pull request by MarcusLongmuir**
This PR adds the classes that are part of the `google-protobuf` library (e.g. `google-protobuf/google/protobuf/any_pb`) that are required by message generated by `protoc`.

These classes were generated by a `protoc` TypeScript plugin that I am working on, with a few of the classes being extended as per the `protoc` JS generator (`any` and `struct` have additional functions). 

I have checked that they match the JS classes.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: None. This is typing more of the library.
- [x] Increase the version number in the header if appropriate. - not appropriate
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: google-protobuf/google-protobuf-tests.ts [hunk 1]
```diff
@@ -1,5 +1,18 @@
 import * as jspb from "google-protobuf";
 
+import "google/protobuf/compiler/plugin_pb";
```
Comment (mhegazy) [inline google-protobuf/google-protobuf-tests.ts]: this does not seem like the module name the file below defines, the file defines `"google-protobuf/google/protobuf/compiler/plugin_pb"`, is that the intention?

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by MarcusLongmuir** [comment]
@mhegazy: How should I structure these additional classes? They're used like this in generated code:
```ts
import * as google_protobuf_timestamp_pb from "google-protobuf/google/protobuf/timestamp_pb";
```

The (current) error on the build is:
```
Error: Directory /home/travis/build/DefinitelyTyped/DefinitelyTyped/google-protobuf/google/protobuf has unused file any_pb.d.ts
```

`any_pb` is used in the test file, but that doesn't seem to solve it. Is the unused file check only checking for usage within the same directory?

**Comment by MarcusLongmuir** [comment]
@minestarks: You looked at the previous `google-protobuf` PRs. Do you have an opinion on how this should be done?

**Comment by MarcusLongmuir** [comment]
@mhegazy: Tests are fixed now 👍

**Pull request closed without merging**